### PR TITLE
Improvements to state synchronization bridge

### PR DESCRIPTION
**Pull request by cygnusv**
* Some minor improvements to the ETH<->Polygon sync contracts
* Deployment script for new coordinator

### CODE REVIEW

File: scripts/deploy_xchain_test.py [hunk 1]
```diff
@@ -22,32 +23,29 @@ def deploy_polygon_contracts(deployer):
...
 
-def main(account_id=None):
+# TODO: Figure out better way to retrieve the TACo app contract address
+def main(taco_app, account_id=None):
     deployer = accounts.load("TGoerli")
     with accounts.use_sender(deployer):
-        root = deploy_eth_contracts(deployer)
-        child, stake_info = deploy_polygon_contracts(deployer)
+        child, _ = deploy_polygon_contracts(deployer)
+        root = deploy_eth_contracts(deployer, child.address, taco_app)
```
Comment (theref) [inline scripts/deploy_xchain_test.py:47]: Do we need taco_app? is this the source?

File: scripts/check_xchain.py [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
+    for i in range(12):
+        time.sleep(60 * i * 5)
+        print("Now: {}".format(time.time()))
+        with networks.polygon.mumbai.use_provider("infura"):
+            stake_info = project.StakeInfo.at("0x96e7dBa88f79e5CCAEBf0c7678539F6C0d719c99")
+            print(
+                stake_info.stakingProviderFromOperator("0xAe87D865F3A507185656aD0ef52a8E0B9f3d58f8")
+            )
+            print(
+                stake_info.stakingProviderFromOperator("0x3B42d26E19FF860bC4dEbB920DD8caA53F93c600")
```
Comment (derekpierre) [inline scripts/check_xchain.py:29]: Where are these addresses obtained from?

File: contracts/contracts/coordination/StakeInfo.sol [hunk 3]
```diff
@@ -20,6 +20,7 @@ contract StakeInfo is AccessControl, IUpdatableStakeInfo, IAccessControlApplicat
     }
 
     constructor(address[] memory updaters) {
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
```
[the remark this answers is not in the document]
Comment (vzotova) [reply contracts/contracts/coordination/StakeInfo.sol:23]: this part moved only to Testnet contracts in following PR, so I'd review it there

**Review by vzotova — APPROVED** [review]
Some changes updated in following PR, considering this - LGTM

File: contracts/xchain/PolygonChild.sol [hunk 4]
```diff
@@ -16,7 +14,6 @@ contract PolygonChild is FxBaseChildTunnel, Ownable {
         address sender,
         bytes memory data
     ) internal override validateSender(sender) {
-        emit MessageReceived(sender, data);
```
[the remark this answers is not in the document]
Comment (cygnusv) [reply contracts/xchain/PolygonChild.sol:19]: The reason was that the FxPortal bridge won't emit such events, so it was pointless. See https://github.com/0xPolygon/fx-portal/blob/296ac8d41579f98d3a4dfb6d41737fae272a30ba/contracts/tunnel/FxBaseChildTunnel.sol#L58

File: scripts/deploy_coordinator_with_fee_model.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        
+        try:
+            stakes = next(d for d in deployments if d["contract_type"] == "StakeInfo")["address"]
+        except StopIteration:
+            raise ValueError("StakeInfo deployment needed")
+
+    # Parameter defaults
+    admin = admin or deployer
+    rate = rate or 1
+    timeout = timeout or 60*60
```
[the remark this answers is not in the document]
Comment (piotr-roslaniec) [reply scripts/deploy_coordinator_with_fee_model.py:71]: Can we put those defaults into `@click.option(..., defaults=...)` or `ape-config.yaml`?

**Comment by cygnusv** [comment]
Merged to `development` by accident, but since #114 was merged to `main` and included these commits anyway, no harm done.

**Pull request merged**